### PR TITLE
Implemented noam scheduler

### PR DESCRIPTION
**Pull request by prubinst**
Previous version of scheduler was producing a different curve. Current code produces a lineal function with peak at 3% of samples, and then cosine decays

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -311,10 +311,21 @@ def _get_data(self, idx, row):
     optimizer = AdamW(model.multi_modal_projector.parameters(), lr=lr)
 else:
     raise ValueError(f"Unknown optimizer: {args.optimizer}")
+
+total_steps = len(train_loader) / args.grad_accum_steps
+print("total steps:", total_steps)
+
+def noam(step):
+    warmup_percent = 0.03
```
Comment (juberti) [inline train.py:319]: suggest storing `int(total_steps * warmup_percent)` as a local so you don't need to recompute it below

**Review by juberti — APPROVED** [review]
